### DISCUSSION TITLE
相同算力、参数下，为什么chatglm3-6b模型会爆内存，而chatglm3-6b-base、chatglm3-6b-32k、chatglm3-6b-128k却能够正常训练？

### DISCUSSION THREAD

**Discussion by Ye-XX** [zai-org/ChatGLM3#1165] [Q&A]
在相同的设备，相同的lora参数下，
chatglm3-6b模型在训练过程中报内存溢出，但是其他模型（chatglm3-6b-base、chatglm3-6b-32k、chatglm3-6b-128k）并不会，都能够正常训练。
甚至chatglm3-6b模型降到更低的参数，也依旧会报内存溢出。
chatglm3-6b模型为什么会需要更大的算力？不是chatglm3-6b-128k需要的算力最大吗？
![111](https://github.com/THUDM/ChatGLM3/assets/42534288/8a53455d-014e-4ecc-8a72-b1b67bb1f988)

这是chatglm3-6b训练时候的参数情况，参数调到很低（只有487,424）。

![6b-2](https://github.com/THUDM/ChatGLM3/assets/42534288/06ec8962-e0b0-46e0-b9d2-b18142dae9ae)

出现的内存溢出情况如下

![6b-3](https://github.com/THUDM/ChatGLM3/assets/42534288/510f0d3b-0c3e-459a-b0b5-00051988d215)

而对于chatglm3-6b-base模型，能够正常运行，训练情况如下，数据量和参数都提升了，也正常运行。

![base-2](https://github.com/THUDM/ChatGLM3/assets/42534288/58e5c54a-c5d1-4ea9-a50f-9f86dd2c07e0)

**Comment by mogumogu2333** [comment]
遇到同样的问题，6b-base可以lora sft, 6b则会CUDA out of memory.
